bmc/power.go

```diff
@@ -0,0 +1,91 @@
+package bmc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// PowerSetter sets the power state of a BMC
+type PowerSetter interface {
+	PowerSet(ctx context.Context, state string) (ok bool, err error)
```

Comment:
Since the `state` string is ambiguous, would it be worth to have pre defined Power state Consts/types, different vendor BMCs could return their own string for Power state, would it be useful if the the provider methods could return the predefined consts?

Reply:
I believe there are trade-offs to using a custom type there. With a custom type, the implementors are now coupled to the interface package. With a string, the implementing package and this interface package aren't coupled. The other side of the trade-off is exactly what you said.

Reply:
some middle ground could be some explicit comments on the interfaces to say something like: implementors should return one of these values. just one idea. I'm sure there are other options out there.

Reply:
add some comments about this idea. let me know what you think?